CHANGELOG.md

```diff
@@ -1,5 +1,85 @@
 # CHANGELOG
 
+## 3.0.0
+
+This release adds pagination to ArsArsenal. In the process of doing this, we've made some breaking changes to the way URLs are constructed. For most users, this upgrade process should be minimal:
+
+### `makeURL` is now `listUrl` and `showUrl`
+
+The old `makeURL` option relied on a null check to determine if the requested url is for a list of items or a single record. To avoid that ambiguity, there are now two endpoints for URL construction:
+
+Instead of:
+
+```javascript
+function makeURL(url, slug) {
+  if (slug == null) {
+    return url
+  } else {
+    return `${url}/${slug}`
+  }
+}
+
+ArsArsenal.render({ makeURL })
+```
+
+Change this to:
+
+```javascript
+let listUrl = url => url
+let showUrl = (url, slug) => `${url}/${slug}`
+
+ArsArsenal.render({ listUrl, showUrl })
+```
+
+**These are the default implementations of each option.** We anticipate
+that this change affects very few users.
+
+### `makeQuery` is now `buildQuery` and returns an object
+
+With pagination, ArsArsenal must now manage multiple query
+parameters. For improved ergonomics, ArsArsenal now builds the query
+string on behalf of the user. Instead of returning a string, return an object of key/value pairs:
+
+Instead of:
+
+```javascript
+function makeQuery(term) {
+  return `q=${term}`
+}
+
+ArsArsenal.render({ makeQuery })
+```
+
+Return an object:
+
+```javascript
+const PAGE_SIZE = 10
+
+function buildQuery({ page, search, sort }) {
+  // Return your pagination/search query implementation:
+  let offset = page * PAGE_SIZE
+  let limit = offset + PAGE_SIZE
+
+  return { q: search, offset, limit, sort }
+}
+```
+
+### Sorting
+
+Adding pagination required us to remove client-side
+sorting. ArsArsenal can't know all of the records on your server, so
+sorting would cause a frustrating reordering of items as new data
+loads.
+
+To enable sorting, take advantage of the `sort` field in the
+`buildQuery` method:
+
+```javascript
+function buildQuery({ page, search, sort }) {
+  return { page, search, sort }
+}
```

Comment:
Is there a fuller example we can provide?

Reply:
Added a more descriptive example and also included it in the readme.